### PR TITLE
mysql-test, part 17

### PR DESCRIPTION
**Pull request by osheroff**
MERGE MANUALLY

- somehow missed NULL as a default value.  scary. 
- allow `REFERENCES( .. )` as a column creation option

@zendesk/rules

### CODE REVIEW

File: src/main/antlr4/imports/mysql_idents.g4 [hunk 1]
```diff
@@ -12,7 +12,7 @@ user_token: (IDENT | QUOTED_IDENT | string_literal);
 
 name: ( id | tokens_available_for_names | INTEGER_LITERAL);
 id: ( IDENT | QUOTED_IDENT );
-literal: (float_literal | integer_literal | string_literal);
+literal: (float_literal | integer_literal | string_literal | NULL);
```
Comment (dadah89) [inline src/main/antlr4/imports/mysql_idents.g4]: does SET DEFAULT support the same values as DEFAULT? https://github.com/zendesk/maxwell/blob/928451d808d7313df92a335dc1c1738e43db3ab0/src/main/antlr4/imports/column_definitions.g4#L90

File: src/main/antlr4/imports/mysql_indices.g4 [hunk 2]
```diff
@@ -47,9 +47,7 @@ index_column_asc_or_desc: ASC | DESC;
 reference_definition:
   REFERENCES table_name
   index_column_list
-  reference_definition_match?
-  reference_definition_on_delete?
-  reference_definition_on_update?
+  ( reference_definition_match | reference_definition_on_delete | reference_definition_on_update)*
```
Comment (dadah89) [inline src/main/antlr4/imports/mysql_indices.g4]: parenthesis spacing doesn't match =)

**Comment by dadah89** [comment]
left few comments, besides that :+1:

File: src/main/antlr4/imports/mysql_idents.g4 (hunk 1 above)
Comment (osheroff) [inline src/main/antlr4/imports/mysql_idents.g4]: it doesn't.  it allows all literals (including TRUE/FALSE, which are not represented here), but not `NOW()` or `CURRENT_TIMESTAMP`

**Pull request merged**